utils/next.ts

```diff
@@ -28,7 +28,15 @@ export const getServerSideCommonProps = async ({
 export const getStaticCommonProps = async ({
   locale,
 }: GetStaticPropsContext) => {
-  const componentTree = await getComponentCategoryGroup()(locale as Locale)
+  const componentTree = (
+    await getComponentCategoryGroup()(locale as Locale)
+  ).filter(({ items }) => items)
+
+  for (const component of componentTree) {
+    if (component.items) {
+      component.items = component.items.filter(({ items }) => items)
+    }
+  }
```

Comment:
How about using the `map` function after `filter`?

```suggestion
const componentTree = (
    await getComponentCategoryGroup()(locale as Locale)
  )
.filter(({ items }) => items)
    .map(component => ({
      ...component,
      items: component.items?.filter(({ items }) => items),
    }))
```